clinica/pipelines/statistics_surface/_outputs.py

```diff
@@ -0,0 +1,226 @@
+    # MATLAB implementation of the Statistics Surface Pipeline
+    # of Clinica.
+    STRUCT_LABELS = {
+        "coefficients": "coef",
+        "TStatistics": "tvaluewithmask",
+        "uncorrectedPValue": "uncorrectedpvaluesstruct",
+        "correctedPValue": "correctedpvaluesstruct",
+        "FDR": "FDR",
+    }
+    for name, result in results.items():
+        _save_to_mat(
+            result,
+            str(filename_root) + "_" + name,
+            STRUCT_LABELS[name],
+            verbose=verbose,
+        )
+
+
+def _save_results_to_bids(results: Dict, filename_root: PathLike, verbose: bool = True):
+    """Write provided results to BIDS format.
+
+    .. warning::
+        This is not implemented yet.
+
+    """
+    warnings.warn("Writing results to BIDS is not implemented yet.")
+
+
+WRITERS = {
+    "json": _save_results_to_json,
+    "mat": _save_results_to_mat,
+    "bids": _save_results_to_bids,
+}
+
+
+def _save_results(
+    results: Dict,
+    filename_root: PathLike,
+    out_formats: Union[str, List] = "all",
+    verbose: bool = True,
+):
+    """Write the provided results to all requested output formats.
+
+    Parameters
+    ----------
+    results : Results to write to disk.
+    filename_root : Filename root common to all output files.
+    out_formats : Either a list of output formats (among "mat", "json", and "bids"), or "all".
+    verbose : Verbose mode.
+    """
+    if out_formats == "all":
+        out_formats = list(WRITERS.keys())
+    for output_format in out_formats:
+        if output_format not in WRITERS:
+            warnings.warn(
+                f"Could not write to {output_format} because writer doesn't exist."
+            )
+        WRITERS[output_format](results, filename_root, verbose=verbose)
+
+
+def _print_clusters(model, threshold: float):
+    """This function prints the results related to total number
+    of clusters, as well as the significative clusters.
+
+    Parameters
+    ----------
+    model : Fitted SLM model.
+    threshold : Cluster defining threshold.
+    """
+    print("#" * 40)
+    print("After correction (Clusterwise Correction for Multiple Comparisons): ")
+    df = model.P["clus"][1]
+    print(df)
+    print(f"Clusters found: {len(df)}")
+    print(f"Significative clusters (after correction): {len(df[df['P'] <= threshold])}")
+
+
+def _plot_stat_map(
+    mesh: np.ndarray,
+    texture: np.ndarray,
+    filename: str,
+    threshold: float = None,
+    title: str = None,
+    verbose: bool = True,
+):
+    """Plot a given texture over the provided mesh using Nilearn's
+    plot_surf_stat_map function.
+
+    Parameters
+    ----------
+    mesh : Mesh of the surface to plot.
+    texture : Texture of the surface to plot.
+    threshold : Threshold to be used for plotting.
+    title : Title to display on the plot.
+    verbose : Verbose mode.
+    """
+    try:
+        from nilearn.plotting import plot_surf_stat_map
```

Comment:
I don't really understand the reason for encapsulating this import in a try catch statement as it is in our dependencies